### ISSUE TITLE
Remove ParametrizedSchedule

### ISSUE THREAD

**Issue by nkanazawa1989** [Qiskit/qiskit-terra#5329]
### What is the expected enhancement?

Now `ParameterizedSchedule` can be unified to `Schedule` because `Schedule` supports parameter object input.
Logic of `InstructionScheduleMap` should be updated accordingly.

**Comment by lcapelluto** [comment]
Changed "Remove" to "Deprecate".

**Comment by lcapelluto** [comment]
This is blocked on https://github.com/Qiskit/qiskit-terra/issues/5333

**Comment by mtreinish** [comment]
Can this be closed now that https://github.com/Qiskit/qiskit-terra/pull/5358 has merged?

**Comment by nkanazawa1989** [comment]
Yes, I think we can merge now. Thanks for the reminder. We still need to remove `ParameterizedSchedule` after few months of next release.
